src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -11,218 +11,232 @@
-                    case WellKnownSynchronizationKind.ParseOptions:
-                    case WellKnownSynchronizationKind.ProjectReference:
-                    case WellKnownSynchronizationKind.SourceGeneratedDocumentIdentity:
-                        return Checksum.Create(value, this);
+                case WellKnownSynchronizationKind.CompilationOptions:
+                case WellKnownSynchronizationKind.ParseOptions:
+                case WellKnownSynchronizationKind.ProjectReference:
+                case WellKnownSynchronizationKind.SourceGeneratedDocumentIdentity:
+                    return Checksum.Create(value, this);
 
-                    case WellKnownSynchronizationKind.MetadataReference:
-                        return Checksum.Create(CreateChecksum((MetadataReference)value, cancellationToken));
+                case WellKnownSynchronizationKind.MetadataReference:
+                    return Checksum.Create(CreateChecksum((MetadataReference)value, cancellationToken));
 
-                    case WellKnownSynchronizationKind.AnalyzerReference:
-                        return Checksum.Create(CreateChecksum((AnalyzerReference)value, cancellationToken));
+                case WellKnownSynchronizationKind.AnalyzerReference:
+                    return Checksum.Create(CreateChecksum((AnalyzerReference)value, cancellationToken));
 
-                    case WellKnownSynchronizationKind.SerializableSourceText:
-                        return Checksum.Create(((SerializableSourceText)value).GetChecksum());
+                case WellKnownSynchronizationKind.SerializableSourceText:
+                    return Checksum.Create(((SerializableSourceText)value).GetChecksum());
 
-                    case WellKnownSynchronizationKind.SourceText:
-                        return Checksum.Create(((SourceText)value).GetChecksum());
+                case WellKnownSynchronizationKind.SourceText:
+                    return Checksum.Create(((SourceText)value).GetChecksum());
 
-                    default:
-                        // object that is not part of solution is not supported since we don't know what inputs are required to
-                        // serialize it
-                        throw ExceptionUtilities.UnexpectedValue(kind);
-                }
+                default:
+                    // object that is not part of solution is not supported since we don't know what inputs are required to
+                    // serialize it
+                    throw ExceptionUtilities.UnexpectedValue(kind);
             }
         }
+    }
 
-        public void Serialize(object value, ObjectWriter writer, SolutionReplicationContext context, CancellationToken cancellationToken)
+    public void Serialize(object value, ObjectWriter writer, SolutionReplicationContext context, CancellationToken cancellationToken)
+    {
+        var kind = value.GetWellKnownSynchronizationKind();
```

Comment:
>var kind = value.GetWellKnownSynchronizationKind(); [](http://example.com/codeflow?start=8&length=51)

There must not be any concern about the serialization and deserialization happening against different versions of this code?

Reply:
There is not. This is just for oop.  Which is always in lockstep.